tests/run_notebook_tests.sh

```
@@ -0,0 +1,45 @@
+cd "$(dirname "$0")/.."
+
+echo "Setting up notebook testing environment..."
+
+# Clean up any existing venv
+if [ -d "test_venv" ]; then
+    rm -rf test_venv
+fi
+
+# Create and activate fresh virtual environment
+python -m venv test_venv
+source test_venv/bin/activate
```

Shellcheck (0.10.0)

[info] Not following: test_venv/bin/activate was not specified as input (see shellcheck -x).

(SC1091)